RecipeApi.Tests/IngredientsTest/IngredientControllerTests.cs

```diff
@@ -18,89 +18,99 @@
-                .UseInMemoryDatabase(databaseName: "GetIngredients_ReturnsOkObjectResult_WithListOfIngredients")
-                .Options;
-
             using (var context = new RecipeDbContext(_options))
             {
                 // Arrange
                 var service = new IngredientService(context, _mapper, _logger);
                 var controller = new IngredientController(service);
 
+                var countOfIngredients = context.Ingredients.Count();
 
-                controller.AddIngredient(new CreateIngredientDto { Name = "Salt" });
-                controller.AddIngredient(new CreateIngredientDto { Name = "Sugar" });
+                controller.AddIngredient(mockIngredient1);
+                controller.AddIngredient(mockIngredient2);
 
                 // Act
                 var result = controller.GetIngredients();
 
                 // Assert
                 var actionResult = Assert.IsType<OkObjectResult>(result.Result);
                 var returnValue = Assert.IsType<List<IngredientDto>>(actionResult.Value);
-                Assert.Equal(2, returnValue.Count);
+                Assert.Equal(countOfIngredients + 2, returnValue.Count);
             }
         }
 
         //get specific ingredient
         [Fact]
-        public void GetIngredient_ReturnsOkObjectResult_WithIngredient()
+        public void GetIngredient_ReturnsOkObjectResult()
         {
-            // Set up in-memory database options
-            DbContextOptions<RecipeDbContext> _options = new DbContextOptionsBuilder<RecipeDbContext>()
-                .UseInMemoryDatabase(databaseName: "GetIngredient_ReturnsOkObjectResult_WithIngredient") //TODO
-                .Options;
-
-            // Arrange
-            var mockIngredient = new CreateIngredientDto { Name = "Salt" };
-
             using (var context = new RecipeDbContext(_options))
             {
                 var service = new IngredientService(context, _mapper, _logger);
                 var controller = new IngredientController(service);
 
-                var addResult = controller.AddIngredient(mockIngredient) as CreatedResult;
+                var result = controller.AddIngredient(mockIngredient1);
+                var addResult=result as CreatedResult;
                 var ingredientId = int.Parse(addResult.Location.Split("/").Last());  //TODO
 
                 // Act
-                var result = controller.GetIngredient(ingredientId);
+                var result1 = controller.GetIngredient(ingredientId);
 
                 // Assert
-                var actionResult = Assert.IsType<OkObjectResult>(result.Result);
+                var actionResult = Assert.IsType<OkObjectResult>(result1.Result);
                 var returnValue = Assert.IsType<IngredientDto>(actionResult.Value);
-                Assert.Equal(ingredientId, returnValue.Id);
+                Assert.Equal(mockIngredient1.Name, returnValue.Name);
```

Comment:
Rename variables to `actual` (your actual value) and `expected` (this is what you expect to get). 
It's better to have `actual.Name` than `mockIngradient1.Name`. I know what to look for.